ads/google/a4a/utils.js

```diff
@@ -589,3 +589,16 @@ export function maybeAppendErrorParameter(adUrl, parameterValue) {
   dev().assert(modifiedAdUrl.length <= MAX_URL_LENGTH);
   return modifiedAdUrl;
 }
+
+/**
+ * Returns a numerical code representing the binary type.
+ * @param {string} type
+ * @return {?number}
+ */
+function getBinaryTypeNumericalCode(type) {
+  return {
+    'production': 0,
+    'control': 1,
+    'canary': 2,
+  }[type] || null;
```

Comment:
This will return null if type == production, is that intended?
console.log('value: ' + ({'production': 0}['production'] || null));
value: null

Reply:
Wasn't intentional, but this should work all the same.

Reply:
Please add a comment indicating the expected behavior.   Would we ever see a situation where the type does not exist in the map?  If not, consider sending a value indicating unknown or async error so it appears in stackdriver logs.

Reply:
This seems like weird behavior that should be fixed. Your googlePageParameters function is going to return an object with {..., 'art': null, ...} for binary type 'production'.

Reply:
Params with null value are ignored and not added to the final URL string. In any case, I've changed the function so that it now returns a string type.